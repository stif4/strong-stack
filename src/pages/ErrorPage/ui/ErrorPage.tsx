import { classNames } from 'shared/helpers/classNames/classNames';
import { useTranslation } from 'react-i18next';
import { Button } from 'shared/ui/Button/Button';
import cls from './ErrorPage.module.scss';

interface ErrorPageProps {
    message?: string;
    className?: string;
}

export const ErrorPage = ({ className, message }: ErrorPageProps) => {
    const { t } = useTranslation();

    const reloadPage = () => {
        location.reload();
    };

    return (
        <div className={classNames(cls.errorPage, {}, [className])}>
            <p>{message ? message : t('Произошла непредвиденная ошибка')}</p>
            <Button onClick={reloadPage}>{t('Обновить страницу')}</Button>
        </div>
    );
};
