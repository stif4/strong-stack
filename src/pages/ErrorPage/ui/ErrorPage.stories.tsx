import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import { ErrorPage } from './ErrorPage';

export default {
    title: 'pages/ErrorPage',
    component: ErrorPage,
} satisfies Meta<typeof ErrorPage>;

const Template: StoryFn<typeof ErrorPage> = (args) => <ErrorPage {...args} />;

export const Light = Template.bind({});
Light.args = {};
