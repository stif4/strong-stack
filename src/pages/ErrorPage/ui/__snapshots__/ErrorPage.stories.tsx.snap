// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/ErrorPage Light smoke-test 1`] = `
<div class="app dark">
  <div class="src-pages-ErrorPage-ui-ErrorPage-module__errorPage">
    <p>
      Произошла непредвиденная ошибка
    </p>
    <button class="src-shared-ui-Button-Button-module__button undefined">
      Обновить страницу
    </button>
  </div>
</div>
`;
